### ISSUE TITLE
Git checkout requires public key be configured with Github

### ISSUE THREAD

**Issue by kingo55** [airbnb/knowledge-repo#57]
Really excited to try this guys - I have been following your project since you posted about it on the blog and I feel like this solves a lot of problems with sharing insights!

Hoping to try it on macOS sierra, the instructions on the readme threw errors each step of the way.

## Step 1 - Git SSH not setup, easier to use HTTPS right?

```bash
$ pip install git+ssh://git@github.com/airbnb/knowledge-repo.git[all]
Collecting git+ssh://git@github.com/airbnb/knowledge-repo.git[all]
  Cloning ssh://git@github.com/airbnb/knowledge-repo.git[all] to /var/folders/b6/vb_3jn_n3019dl8tc78wv2cw0000gn/T/pip-pIGqKj-build
Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Command "git clone -q ssh://git@github.com/airbnb/knowledge-repo.git[all] /var/folders/b6/vb_3jn_n3019dl8tc78wv2cw0000gn/T/pip-pIGqKj-build" failed with error code 128 in None
```

Easier to use HTTPS right?:

```bash
$ pip install git+https://github.com/airbnb/knowledge-repo.git
```

Then again, judging from error messages down the track, I suspect Git SSH is mandatory elsewhere?

## Step 2 - Should point to some basic instructions for setting up the Git repo

Running this throws an error. 

```bash
$ knowledge_repo --repo ./example_repo init
Traceback (most recent call last):
  File "/Users/robertkingston/Documents/knowledge-repo/venv/bin/knowledge_repo", line 207, in <module>
    kr = knowledge_repo.KnowledgeRepository.create_for_uri(args.repo, embed_tooling=embed_tooling)
  File "/Users/robertkingston/Documents/knowledge-repo/venv/lib/python2.7/site-packages/knowledge_repo/repository.py", line 65, in create_for_uri
    return cls._get_subclass_for(scheme).create(uri, **kwargs)
  File "/Users/robertkingston/Documents/knowledge-repo/venv/lib/python2.7/site-packages/knowledge_repo/repositories/gitrepository.py", line 51, in create
    sm = repo.create_submodule(name='embedded_knowledge_repo', path='.resources', url=tooling_repo, branch=tooling_branch)
  File "/Users/robertkingston/Documents/knowledge-repo/venv/lib/python2.7/site-packages/git/repo/base.py", line 306, in create_submodule
    return Submodule.add(self, *args, **kwargs)
  File "/Users/robertkingston/Documents/knowledge-repo/venv/lib/python2.7/site-packages/git/objects/submodule/base.py", line 389, in add
    mrepo = cls._clone_repo(repo, url, path, name, **kwargs)
  File "/Users/robertkingston/Documents/knowledge-repo/venv/lib/python2.7/site-packages/git/objects/submodule/base.py", line 251, in _clone_repo
    clone = git.Repo.clone_from(url, module_checkout_path, **kwargs)
  File "/Users/robertkingston/Documents/knowledge-repo/venv/lib/python2.7/site-packages/git/repo/base.py", line 966, in clone_from
    return cls._clone(git, url, to_path, GitCmdObjectDB, progress, **kwargs)
  File "/Users/robertkingston/Documents/knowledge-repo/venv/lib/python2.7/site-packages/git/repo/base.py", line 912, in _clone
    finalize_process(proc, stderr=stderr)
  File "/Users/robertkingston/Documents/knowledge-repo/venv/lib/python2.7/site-packages/git/util.py", line 155, in finalize_process
    proc.wait(**kwargs)
  File "/Users/robertkingston/Documents/knowledge-repo/venv/lib/python2.7/site-packages/git/cmd.py", line 335, in wait
    raise GitCommandError(self.args, status, errstr)
git.exc.GitCommandError: 'git clone -b master --separate-git-dir=/Users/robertkingston/Documents/knowledge-repo/example_repo/.git/modules/embedded_knowledge_repo -v git@github.com:airbnb/knowledge-repo.git /Users/robertkingston/Documents/knowledge-repo/example_repo/.resources' returned with exit code 128
stderr: 'Cloning into '/Users/robertkingston/Documents/knowledge-repo/example_repo/.resources'...
Warning: Permanently added the RSA host key for IP address 'x.x.x.x' to the list of known hosts.
Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
'
```

Advice on setting up a free private git repo would be helpful for less technical users like myself. e.g.:

https://confluence.atlassian.com/bitbucket/create-a-git-repository-759857290.html

This might be good to add to the start of the guide as a "Prerequisites" list, along with a notice at step 2.

**Comment by matthewwardrop** [comment]
Greetings Robert! Thanks for your interest in this project!

You're quite right. Using https makes a lot more sense, and it is at no point actually necessary to use the ssh protocol with git. The error messages seem to suggest you haven't set up a public key with Github; but this should not be necessary. I'll fix this asap.

**Comment by matthewwardrop** [comment]
Thanks again for your enthusiasm and interest Robert. This is is solved by #61 , which will be merged imminently. Let us know if you come across any other difficulties.
